File: test/samples/keywords/keywords.pas
program Keywords;

type
    { may conflict in c# }
    Console = record
        A: longint; { the first letter of the alphabet }
        Static: longint; { an integer }
    end;

    { may conflict in c# }
    System_ = record
        Return: longint; { not the end of the function }
        Void: array of longint; { not nothing }
    end;

    { not the main function }
    Main = record
        Int: System_; { not an integer }
        IfTrue: longint; { should not cause conflict }
    end;

    T_For_ = array of longint;
    T_Words = array of Main;

{ @param If_ not a condition }
{ @param Class not a class }
{ @param I just a string }
{ @param In_ not in }
{ @param For_ not a loop }
{ @param Words contains lots of things }
procedure Keywords(If_: longint; Class: char; const I: string; const In_: Console; const For_: T_For_; const Words: T_Words);
begin
    {* TODO If this compiles, it is already a good step! *}
end;

var
    If_: longint; { not a condition }
    Class: char; { not a class }
    I: string; { just a string }
    In_: Console; { not in }
    For_: T_For_; { not a loop }
    Words: T_Words; { contains lots of things }
    j, k: longint;
begin
    readln(If_);
    readln(Class);
    readln(I);
    readln(In_.A, In_.Static);
    setLength(For_, If_);
    for j := 0 to If_ - 1 do
        read(For_[j]);
    readln();
    setLength(Words, 2);
    for j := 0 to 2 - 1 do
    begin
        readln(Words[j].Int.Return);
        setLength(Words[j].Int.Void, 3);
        for k := 0 to 3 - 1 do
            read(Words[j].Int.Void[k]);
        readln();
        readln(Words[j].IfTrue);
    end;
    Keywords(If_, Class, I, In_, For_, Words);
end.

File: test/samples/keywords/keywords.cpp
#include <iostream>
#include <istream>
#include <string>
#include <vector>

/// may conflict in c#
struct Console {
    int a; ///< the first letter of the alphabet
    int static_; ///< an integer
};

/// may conflict in c#
struct System {
    int return_; ///< not the end of the function
    std::vector<int> void_; ///< not nothing
};

/// not the main function
struct Main {
    System int_; ///< not an integer
    int if_true; ///< should not cause conflict
};

/// \param if_ not a condition
/// \param class_ not a class
/// \param i just a string
/// \param in not in
/// \param for_ not a loop
/// \param words contains lots of things
/// \param words_1 an integer
void keywords(int if_, char class_, const std::string& i, const Console& in, const std::vector<int>& for_, const std::vector<Main>& words, int words_1) {
    /* TODO If this compiles, it is already a good step! */
}

int main() {
    int if_; ///< not a condition
    std::cin >> if_;
    char class_; ///< not a class
    std::cin >> class_;
    std::string i; ///< just a string
    std::getline(std::cin >> std::ws, i);
    Console in; ///< not in
    std::cin >> in.a >> in.static_;
    std::vector<int> for_(if_); ///< not a loop
    for (int& j : for_)
        std::cin >> j;
    std::vector<Main> words(2); ///< contains lots of things
    for (Main& j : words) {
        std::cin >> j.int_.return_;
        j.int_.void_.resize(3);
        for (int& k : j.int_.void_)
            std::cin >> k;
        std::cin >> j.if_true;
    }
    int words_1; ///< an integer
    std::cin >> words_1;
    keywords(if_, class_, i, in, for_, words, words_1);
}
